File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const n = priorities.length;
    let pivot = 0;
    
    for (let i = 0; i < n; i++) {
        const max = Math.max(...priorities);
        
        while (priorities[pivot % n] < max) pivot++;
        
        priorities[pivot % n] = 0;
        
        if (pivot % n === location) return i + 1;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    n = len(priorities)
    pivot = 0
    
    for i in range(n):
        _max = max(priorities)
        
        while priorities[pivot % n] < _max:
            pivot += 1
        
        priorities[pivot % n] = 0
        
        if pivot % n == location:
            return i + 1
